packages/events-example/src/commands/reconcile/index.ts

```diff
@@ -1,188 +1,136 @@
-    ).reduce(async (accP, stack) => {
-      const acc = await accP;
-      const fullStack = await this.getStack(stack.name, pulumiProgram({}));
-      const info = await fullStack.info();
-      if (info?.kind != 'destroy') {
-        acc[stack.name] = stack;
-        this.debug('adding stack', stack);
-      }
-      return acc;
-    }, Promise.resolve({} as { [key: string]: StackSummary }));
-    this.debug(stacks);
-    return stacks;
-  }
-
-  async loadInstances(
-    organization: string,
-    entity: string
-  ): Promise<{ [key: string]: Instance }> {
-    const instances = (
-      await this.nile.entities.listInstances({
-        org: organization,
-        type: entity,
-      })
-    ).reduce((acc, instance: Instance) => {
-      if (instance) {
-        acc[instance.id] = instance;
-      }
+    if (flags.statusCheckOnly) {
+      console.log({ stacks, instances });
+      return { stacks, instances };
+    }
 
-      return acc;
-    }, {} as { [key: string]: Instance });
-    this.debug(instances);
-    return instances;
+    this.synchronizeDataPlane(instances, stacks);
+    this.listenForNileEvents(flags.entity, this.findLastInstance(Object.values(instances)));
   }
 
-  async run(): Promise<any> {
-    const { flags } = await this.parse(Reconcile);
-
-    // pulumi setup
-    this.localWorkspace = await LocalWorkspace.create({
-      projectSettings: { name: flags.workspace, runtime: 'nodejs' },
-    });
-    this.localWorkspace.installPlugin('aws', 'v4.0.0');
-
-    // nile setup
-    this.nile = Nile({ basePath: flags.basePath, workspace: flags.workspace });
+  async connectNile(parsedFlags: {[name: string]: any}) {
+    this.nile = Nile({ basePath: parsedFlags.basePath, workspace: parsedFlags.workspace });
     const token = await this.nile.developers
       .loginDeveloper({
         loginInfo: {
-          email: flags.email,
-          password: flags.password,
+          email: parsedFlags.email,
+          password: parsedFlags.password,
         },
       })
       .catch((error: unknown) => {
         // eslint-disable-next-line no-console
-        console.error(error);
+        console.error("Nile authentication failed", error);
       });
     this.nile.authToken = token?.token;
+  }
 
-    // load our data
-    const stacks = await this.loadPulumiStacks();
-    const instances = await this.loadInstances(
-      flags.organization,
-      flags.entity
-    );
-    let seq = 0;
-    for (const instance of Object.values(instances)) {
-      if (seq == null || (instance?.seq || seq) > seq) {
-        seq = instance?.seq || seq;
-      }
-    }
+  async loadNileInstances(organization: string, entity: string): Promise<{ [key: string]: Instance }> {
+    const instances = (
+      await this.nile.entities.listInstances({
+        org: organization,
+        type: entity,
+      })
+    )
+      .filter((value: Instance) => value !== null && value !== undefined)
+      .reduce((acc, instance: Instance) => {
```

Comment:
since filter was added, this should probably be a `map` for clarity, since we originally did it in the reduce.